core/src/components/datetime-button/datetime-button.tsx

```diff
@@ -0,0 +1,411 @@
+import type { ComponentInterface } from '@stencil/core';
+import { Component, Element, Host, Prop, State, h } from '@stencil/core';
+
+import { getIonMode } from '../../global/ionic-global';
+import type { Color, DatetimePresentation, DatetimeParts } from '../../interface';
+import { componentOnReady, addEventListener } from '../../utils/helpers';
+import { printIonError, printIonWarning } from '../../utils/logging';
+import { createColorClasses } from '../../utils/theme';
+import { getToday } from '../datetime/utils/data';
+import { getMonthAndYear, getMonthDayAndYear, getLocalizedDateTime, getLocalizedTime } from '../datetime/utils/format';
+import { is24Hour } from '../datetime/utils/helpers';
+import { parseDate } from '../datetime/utils/parse';
+/**
+ * @virtualProp {"ios" | "md"} mode - The mode determines which platform styles to use.
+ *
+ * @slot date-target - Content displayed inside of the date button.
+ * @slot time-target - Content displayed inside of the time button.
+ *
+ * @part native - The native HTML button that wraps the slotted text.
+ */
+@Component({
+  tag: 'ion-datetime-button',
+  styleUrl: 'datetime-button.scss',
```

Comment:
We will need to specify the stylesheet for both modes here, otherwise manually setting the modes directly on the element will have no effect.

i.e.:
```ts
styleUrls: {
  ios: 'datetime-button.scss',
  md: 'datetime-button.scss'
}
```

You can confirm it is working by inspecting the element after making this change and observing if the `ios` class is reflected when in Chrome without device emulation. 

Also after making this change, the ripple effect will not be applied when you are manually setting `mode="ios"`.